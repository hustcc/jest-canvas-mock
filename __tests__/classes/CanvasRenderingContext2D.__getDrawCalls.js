let ctx;
beforeEach(() => {
  // get a new context each test
  ctx = document.createElement('canvas').getContext('2d');
});

const img = new Image();
img.src = 'https://placekitten.com/400/300';
img.width = 400;
img.height = 300;

const path = new Path2D();
path.arc(100, 101, 10, 0, Math.PI * 2);

afterEach(() => {
  const drawCalls = ctx.__getDrawCalls();
  expect(drawCalls).toMatchSnapshot();
});

describe('__getDrawCalls', () => {
  it('should have a draw call when clearRect is called', () => {
    ctx.clearRect(1, 2, 3, 4);
  });

  it('should not have a draw call when clearRect is passed bad values', () => {
    ctx.clearRect(NaN, 1, 2, 3);
  });

  it('should have a draw call when fillRect is called', () => {
    ctx.fillRect(1, 2, 3, 4);
  });

  it('should not have a draw call when fillRect is passed bad values', () => {
    ctx.fillRect(NaN, 1, 2, 3);
  });

  it('should have a draw call when strokeRect is called', () => {
    ctx.strokeRect(1, 2, 3, 4);
  });

  it('should not have a draw call when strokeRect is passed bad values', () => {
    ctx.strokeRect(NaN, 1, 2, 3);
  });

  it('should have a draw call when drawImage is called', () => {
    ctx.drawImage(img, 0, 0);
  });

  it('should not have a draw call when drawImage is called with non-finite numbers', () => {
    ctx.drawImage(img, NaN, 0);
  });

  it('should have a draw call when drawImage is called with size parameters', () => {
    ctx.drawImage(img, 0, 0, 100, 100);
  });

  it('should not have a draw call when drawImage is called with non-finite numbers and size parameters', () => {
    ctx.drawImage(img, NaN, 0, 100, 100);
  });

  it('should have a draw call when drawImage is called with source parameters', () => {
    ctx.drawImage(img, 0, 0, 100, 100, 0, 0, 200, 200);
  });

  it('should not have a draw call when drawImage is called with non-finite numbers and source parameters', () => {
    ctx.drawImage(img, NaN, 0, 100, 100, 0, 0, 200, 200);
  });

  it('should have a draw call when fill() is called', () => {
    ctx.beginPath();
    ctx.arc(100, 101, 10, 0, Math.PI * 2);
    ctx.fill();
  });

  it('should have a draw call when fill() is called with a fillRule', () => {
    ctx.beginPath();
    ctx.arc(100, 101, 10, 0, Math.PI * 2);
    ctx.fill('evenodd');
  });

  it('should have a draw call when using a path', () => {
    ctx.fill(path);
  });

  it('should have a draw call when fillRule is valid', () => {
    ctx.fill(path, 'evenodd');
  });

  it('should have a draw call when fillText is valid', () => {
    ctx.fillText('hello world', 0, 0);
  });

  it('should have a draw call when fillText has valid maxWidth', () => {
    ctx.fillText('hello world', 0, 0, 100);
  });

  it('should not have a draw call when fillText is not valid', () => {
    ctx.fillText('hello world', 0, NaN);
  });

  it('should not have a draw call when fillText is not valid', () => {
    ctx.fillText('hello world', 0, 0, NaN);
  });

  it('should have a draw call when strokeText is valid', () => {
    ctx.strokeText('hello world', 0, 0);
  });

  it('should have a draw call when strokeText has valid maxWidth', () => {
    ctx.strokeText('hello world', 0, 0, 100);
  });

  it('should not have a draw call when strokeText is not valid', () => {
    ctx.strokeText('hello world', 0, NaN);
  });

  it('should not have a draw call when strokeText is not valid', () => {
    ctx.strokeText('hello world', 0, 0, NaN);
  });

  it('should have a draw call when stroke is called', () => {
    ctx.beginPath();
    ctx.arc(100, 101, 10, 0, Math.PI * 2);
    ctx.stroke();
  });

  it('should have a draw call when stroke is called with a path', () => {
    ctx.stroke(path);
  });
});
