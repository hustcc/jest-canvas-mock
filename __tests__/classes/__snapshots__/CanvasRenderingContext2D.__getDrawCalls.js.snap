// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__getDrawCalls should have a draw call when clearRect is called 1`] = `
Array [
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "clearRect",
  },
]
`;

exports[`__getDrawCalls should have a draw call when drawImage is called 1`] = `
Array [
  Object {
    "props": Object {
      "dHeight": 300,
      "dWidth": 400,
      "dx": 0,
      "dy": 0,
      "img": <img
        height="300"
        src="https://placekitten.com/400/300"
        width="400"
      />,
      "sHeight": 300,
      "sWidth": 400,
      "sx": 0,
      "sy": 0,
    },
    "type": "drawImage",
  },
]
`;

exports[`__getDrawCalls should have a draw call when drawImage is called with size parameters 1`] = `
Array [
  Object {
    "props": Object {
      "dHeight": 300,
      "dWidth": 400,
      "dx": 0,
      "dy": 0,
      "img": <img
        height="300"
        src="https://placekitten.com/400/300"
        width="400"
      />,
      "sHeight": 300,
      "sWidth": 400,
      "sx": 0,
      "sy": 0,
    },
    "type": "drawImage",
  },
]
`;

exports[`__getDrawCalls should have a draw call when drawImage is called with source parameters 1`] = `
Array [
  Object {
    "props": Object {
      "dHeight": 200,
      "dWidth": 200,
      "dx": 0,
      "dy": 0,
      "img": <img
        height="300"
        src="https://placekitten.com/400/300"
        width="400"
      />,
      "sHeight": 100,
      "sWidth": 100,
      "sx": 0,
      "sy": 0,
    },
    "type": "drawImage",
  },
]
`;

exports[`__getDrawCalls should have a draw call when fill() is called 1`] = `
Array [
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {},
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "fill",
  },
]
`;

exports[`__getDrawCalls should have a draw call when fill() is called with a fillRule 1`] = `
Array [
  Object {
    "props": Object {
      "fillRule": "evenodd",
      "path": Array [
        Object {
          "props": Object {},
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "fill",
  },
]
`;

exports[`__getDrawCalls should have a draw call when fillRect is called 1`] = `
Array [
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "fillRect",
  },
]
`;

exports[`__getDrawCalls should have a draw call when fillRule is valid 1`] = `
Array [
  Object {
    "props": Object {
      "fillRule": "evenodd",
      "path": Array [
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "fill",
  },
]
`;

exports[`__getDrawCalls should have a draw call when fillText has valid maxWidth 1`] = `
Array [
  Object {
    "props": Object {
      "maxWidth": 100,
      "text": "hello world",
      "x": 0,
      "y": 0,
    },
    "type": "fillText",
  },
]
`;

exports[`__getDrawCalls should have a draw call when fillText is valid 1`] = `
Array [
  Object {
    "props": Object {
      "maxWidth": null,
      "text": "hello world",
      "x": 0,
      "y": 0,
    },
    "type": "fillText",
  },
]
`;

exports[`__getDrawCalls should have a draw call when stroke is called 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "stroke",
  },
]
`;

exports[`__getDrawCalls should have a draw call when stroke is called with a path 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "stroke",
  },
]
`;

exports[`__getDrawCalls should have a draw call when strokeRect is called 1`] = `
Array [
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "strokeRect",
  },
]
`;

exports[`__getDrawCalls should have a draw call when strokeText has valid maxWidth 1`] = `
Array [
  Object {
    "props": Object {
      "maxWidth": 100,
      "text": "hello world",
      "x": 0,
      "y": 0,
    },
    "type": "strokeText",
  },
]
`;

exports[`__getDrawCalls should have a draw call when strokeText is valid 1`] = `
Array [
  Object {
    "props": Object {
      "maxWidth": null,
      "text": "hello world",
      "x": 0,
      "y": 0,
    },
    "type": "strokeText",
  },
]
`;

exports[`__getDrawCalls should have a draw call when using a path 1`] = `
Array [
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "fill",
  },
]
`;

exports[`__getDrawCalls should not have a draw call when clearRect is passed bad values 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when drawImage is called with non-finite numbers 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when drawImage is called with non-finite numbers and size parameters 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when drawImage is called with non-finite numbers and source parameters 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when fillRect is passed bad values 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when fillText is not valid 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when fillText is not valid 2`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when strokeRect is passed bad values 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when strokeText is not valid 1`] = `Array []`;

exports[`__getDrawCalls should not have a draw call when strokeText is not valid 2`] = `Array []`;
